Kernel/Syscalls/write.cpp

```diff
@@ -72,7 +75,7 @@ KResultOr<ssize_t> Process::do_write(FileDescription& description, const UserOrK
     }
 
     while ((size_t)total_nwritten < data_size) {
-        if (!description.can_write()) {
+        while (!description.can_write()) {
```

Comment:
This is needed for the case where I try to run 2 instances of `yes > /dev/hvc0`. If it blocks, then both of the processes get unblocked when the virtioqueue gets unblocked (it calls evaluate_block_conditions, which wakes up all blocked threads). I'm not sure how other device drivers deal with this problem, from what I can see in their implementations it looks like it could end up returning 0 from a write syscall, which isn't allowed by POSIX.

Reply:
I'm going to fix this in a separate PR https://github.com/SerenityOS/serenity/pull/7011